src/audio/copier/copier.c

```diff
@@ -610,25 +610,23 @@ static int copier_prepare(struct comp_dev *dev)
 	if (ret == COMP_STATUS_STATE_ALREADY_SET)
 		return PPL_STATUS_PATH_STOP;
 
-	if (cd->endpoint_num) {
+	for (i = 0; i < cd->endpoint_num; i++) {
+		ret = cd->endpoint[i]->drv->ops.prepare(cd->endpoint[i]);
+		if (ret < 0)
+			break;
```

Comment:
This has a slight behaviour change. If error happens, before we skipped running prepare on pre-patch L616, but now we still. Is this intentional? If not, check on "&& !ret" on L619 will do.

Reply:
@kv2019i sorry, not sure why you think the behaviour is changed here - I don't see it, but I'd agree that it would be better to avoid unnecessary changes. Just remove lines 629-631, don't change anything else

Reply:
@lyakh Ack, the scenario I have above cannot happen. If we execute prepare(), endpoint_num must be non-zero, so we never go to the branch starting at L619. So ok, this is good.